test/e2e/standard/sanitychecker.go

```diff
@@ -66,6 +70,27 @@ func NewSanityChecker(ctx context.Context, log *logrus.Entry, cs *internalapi.Op
 	if err != nil {
 		return nil, err
 	}
+
+	scc.RealRPClient, err = azure.NewClientFromEnvironment(true, true)
+	log.Debugf("New RealRP client ended with %v", scc.RealRPClient)
+	if err != nil {
+		log.Error(err)
+	}
+
+	scc.FakeRPClient, err = azure.NewClientFromEnvironment(true, false)
+	log.Debugf("New FakeRP client ended with %v", scc.RealRPClient)
+	if err != nil {
+		log.Error(err)
+	}
+	if scc.RealRPClient == nil && scc.FakeRPClient == nil {
+		log.Error("Unable to provision either of the azure clients")
+		return nil, err
+	}
+	if scc.RealRPClient != nil {
```

Comment:
rationale for removing AnyRPClient is that this logic isn't right, but I don't know what the right logic looks like still

Reply:
Removed AnyRP for now. I think a test dial has to be performed to decide if an Azure Client is valid.